slay/truss_adapter/deploy.py

```diff
@@ -0,0 +1,325 @@
+
+    config.write_to_yaml_file(
+        truss_dir / serving_image_builder.CONFIG_FILE, verbose=False
+    )
+
+    # Copy other sources.
+    model_dir = truss_dir / truss_config.DEFAULT_MODEL_MODULE_DIR
+    model_dir.mkdir(parents=True, exist_ok=True)
+    shutil.copy(
+        processor_dir / f"{definitions.PROCESSOR_MODULE}.py",
+        model_dir / _MODEL_CLASS_FILENAME,
+    )
+
+    pkg_dir = truss_dir / truss_config.DEFAULT_BUNDLED_PACKAGES_DIR
+    pkg_dir.mkdir(parents=True, exist_ok=True)
+    if maybe_stub_file is not None:
+        shutil.copy(maybe_stub_file, pkg_dir)
+    # TODO This assume all imports are absolute w.r.t workflow root (or site-packages).
+    # Also: apparently packages need an `__init__`, or crash.
+    _copy_python_source_files(workflow_root, pkg_dir / pkg_dir)
+
+    # TODO Truss package contains this from `{ include = "slay", from = "." }`
+    # pyproject.toml. But for quick dev loop just copy from local.
+    shutil.copytree(
+        os.path.dirname(slay.__file__),
+        pkg_dir / "slay",
+        dirs_exist_ok=True,
+    )
+    return truss_dir
+
+
+def get_api_key_from_truss_config() -> str:
+    return remote_factory.load_config().get("baseten", "api_key")
+
+
+class _BasetenEnv(enum.Enum):
+    LOCAL = enum.auto()
+    STAGING = enum.auto()
+    PROD = enum.auto()
+    DEV = enum.auto()
+
+
+def _infer_env(baseten_url: str) -> _BasetenEnv:
+    if baseten_url in {"localhost", "127.0.0.1", "0.0.0.0"}:
+        return _BasetenEnv.LOCAL
+
+    if "staging" in baseten_url:
+        return _BasetenEnv.STAGING
+
+    if "dev" in baseten_url:
+        return _BasetenEnv.DEV
+
+    return _BasetenEnv.PROD
+
+
+def _model_url(baseten_env: _BasetenEnv, model_id: str) -> str:
+    if baseten_env == _BasetenEnv.LOCAL:
+        return f"http://localhost:8000/models/{model_id}"
+
+    if baseten_env == _BasetenEnv.STAGING:
+        return f"https://app.staging.baseten.co/models/{model_id}"
+
+    if baseten_env == _BasetenEnv.DEV:
+        return f"https://app.dev.baseten.co/models/{model_id}"
+
+    return f"https://model-{model_id}.api.baseten.co/production"
+
+
+class BasetenClient:
+    """Helper to deploy models on baseten and inquire their status."""
+
+    # TODO: use rest APIs where possible in stead of graphql_query.
+    def __init__(self, baseten_url: str, baseten_api_key: str) -> None:
+        self._baseten_url = baseten_url
+        self._baseten_env = _infer_env(baseten_url)
+        self._baseten_api_key = baseten_api_key
+        self._remote_provider: truss_remote.TrussRemote = self._create_remote_provider()
+
+    def deploy_truss(self, truss_root: Path) -> definitions.BasetenRemoteDescriptor:
+        tr = truss.load(str(truss_root))
+        model_name = tr.spec.config.model_name
+        assert model_name is not None
+
+        logging.info(f"Deploying model `{model_name}`.")
+        service = self._remote_provider.push(
+            tr, model_name=model_name, trusted=True, publish=False
```

Comment:
worth adding a comment that trusted is required (for api key) and a TODO for setting publish properly to support dev loop or promote down the line.